File: js-pr/56/javascript/code-quality-error-handling/database_connector_33.js
// {fact rule=code-quality-error-handling@v1.0 defects=0}
function connectDatabase(connectionString) {
    const mysql = require('mysql');
    try {
        const connection = mysql.createConnection(connectionString);
        connection.connect((err) => {
            if (err) {
                console.error('Database connection error:', err.message);
                return null;
            }
        });
        return connection;
    } catch (error) {
        console.error('Database setup error:', error.message);
        return null;
    }
}
// {/fact}

File: python-pr/52/python/code-quality-error-handling/database_connector_33.py
# {fact rule=code-quality-error-handling@v1.0 defects=0}
def connect_db(connection_string):
    import sqlite3
    try:
        conn = sqlite3.connect(connection_string)
        return conn
    except sqlite3.Error as e:
        print(f"Database error: {e}")
        return None
# {/fact}